src/server/node.rs

```diff
@@ -211,13 +211,23 @@ impl<C> Node<C>
         let store = self.store.clone();
         let ch = event_loop.channel();
 
+        let (tx, rx): (mpsc::Sender<i32>, mpsc::Receiver<i32>) = mpsc::channel();
         let builder = thread::Builder::new().name(thd_name!(format!("raftstore-{}", store_id)));
         let h = try!(builder.spawn(move || {
             let mut store = Store::new(ch, store, cfg, db, trans, pd_client, snap_mgr).unwrap();
+            let result = store.init();
+            // make sure the initialization of store is finished (even if error happens) in startup
+            tx.send(0).unwrap();
+            if let Err(e) = result {
+                error!("store {} prepare err {:?}", store_id, e);
```

Comment:
Sure. Panic could be better consider the logic correctness (forget the above I mentioned about compatibility with the old code).